src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -370,16 +393,26 @@ export class OsdListComponent extends ListWithDetails implements OnInit {
     }
   }
 
+  filterIndivFlags(flags: string[]) {
+    return flags.filter((f) => this.indivFlagNames.includes(f));
+  }
+
+  filterClusterWideFlags(flags: string[]) {
+    return flags.filter((f) => !this.disabledFlags.includes(f));
+  }
+
   getOsdList() {
-    this.osdService.getList().subscribe((data: any[]) => {
-      this.osds = data.map((osd) => {
+    const observables = [this.osdService.getList(), this.osdService.getFlags()];
+    observableForkJoin(observables).subscribe((resp: [any[], string[]]) => {
+      this.osds = resp[0].map((osd) => {
         osd.collectedStates = OsdListComponent.collectStates(osd);
         osd.stats_history.out_bytes = osd.stats_history.op_out_bytes.map((i: string) => i[1]);
         osd.stats_history.in_bytes = osd.stats_history.op_in_bytes.map((i: string) => i[1]);
         osd.stats.usage = osd.stats.stat_bytes_used / osd.stats.stat_bytes;
         osd.cdIsBinary = true;
         return osd;
       });
+      this.clusterWideFlags = resp[1];
```

Comment:
Following the previous comment:
```suggestion
      this.clusterWideFlags = resp[1];
      this.osds = resp[0].map((osd) => {
        osd.collectedStates = OsdListComponent.collectStates(osd);
        osd.stats_history.out_bytes = osd.stats_history.op_out_bytes.map((i: string) => i[1]);
        osd.stats_history.in_bytes = osd.stats_history.op_in_bytes.map((i: string) => i[1]);
        osd.stats.usage = osd.stats.stat_bytes_used / osd.stats.stat_bytes;
        osd.cdIsBinary = true;
        osd.cdIndivFlags = osd.state.filter((f: string) => this.indivFlagNames.includes(f));
        osd.cdClusterFlags = this.clusterWideFlags.filter(
          (f: string) => !this.disabledFlags.includes(f)
        );
        return osd;
      });
```